packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -239,23 +240,51 @@ export default class VTable {
                 break;
 
             case TableOperation.DeleteRow:
-                this.forEachCellOfCurrentRow((cell, i) => {
-                    let nextCell = this.getCell(this.row + 1, i);
+                const deleteRowHandler = (cell: VCell, i: number, rowIndex: number) => {
+                    let nextCell = this.getCell(rowIndex + 1, i);
                     if (cell.td && cell.td.rowSpan > 1 && nextCell.spanAbove) {
                         nextCell.td = cell.td;
                     }
-                });
-                this.cells.splice(this.row, 1);
+                };
+                if (this.selection) {
+                    const { firstCell, lastCell } = this.selection;
+                    for (let rowIndex = firstCell.y; rowIndex <= lastCell.y; rowIndex++) {

```

Comment:
curious, do we need to check lastCell.y >= firstCell.y? Otherwise it is possible to go into infinite loop

Reply:
I don't think it is possible. The first selected cell will be ways the 0,0, so the coordinate of the other cells will be always bigger. 

`export function getCellCoordinates(vTable: VTable, cellInput: Node): Coordinates {
    let result: Coordinates;
    if (vTable?.cells) {
        for (let indexY = 0; indexY < vTable.cells.length; indexY++) {
            for (let indexX = 0; indexX < vTable.cells[indexY].length; indexX++) {
                if (cellInput == vTable.cells[indexY][indexX].td) {
                    result = {
                        x: indexX,
                        y: indexY,
                    };
                }
            }
        }
    }

    return result;
}`